disperser/apiserver/server.go

```diff
@@ -98,6 +98,10 @@ func NewDispersalServer(
 }
 
 func (s *DispersalServer) DisperseBlobAuthenticated(stream pb.Disperser_DisperseBlobAuthenticatedServer) error {
+
+	ctx, cancel := context.WithTimeout(stream.Context(), s.serverConfig.GrpcTimeout)
```

Comment:
should this create a brand new context, with `deadline = min{stream.Context().Deadline(), s.serverConfig.GrpcTimeout)`?

Reply:
It seems that `WithTimeout` already performs this check: 

```
func WithDeadlineCause(parent Context, d time.Time, cause error) (Context, CancelFunc) {
	if parent == nil {
		panic("cannot create context from nil parent")
	}
	if cur, ok := parent.Deadline(); ok && cur.Before(d) {
		// The current deadline is already sooner than the new one.
		return WithCancel(parent)
	}
```